### PR TITLE
Check if blobs are expired before returning

### PR DESCRIPTION
**Pull request by mooselumph**
## Checks

- [ ] I've made sure the lint is passing in this PR.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: disperser/apiserver/server.go [hunk 1]
```diff
@@ -727,6 +727,12 @@ func (s *DispersalServer) RetrieveBlob(ctx context.Context, req *pb.RetrieveBlob
 		return nil, api.NewInternalError("failed to get blob metadata, please retry")
 	}
 
+	if blobMetadata.Expiry < uint64(time.Now().Unix()) {
+		s.metrics.HandleNotFoundRpcRequest("RetrieveBlob")
+		s.metrics.HandleNotFoundRequest("RetrieveBlob")
+		return nil, api.NewNotFoundError("blob expired")
```
Comment (ian-shim) [inline disperser/apiserver/server.go:733]: should we use the same error message as L723? 
Users may be relying on the error message to handle different failure cases

Comment (mooselumph) [reply disperser/apiserver/server.go:733]: Yeah, I was wondering about this. I'll change it back for now, but we should probably revisit error message stability in the future.

**Review by ian-shim — APPROVED** [review]

File: disperser/apiserver/server.go [hunk 2]
```diff
@@ -727,6 +727,12 @@ func (s *DispersalServer) RetrieveBlob(ctx context.Context, req *pb.RetrieveBlob
 		return nil, api.NewInternalError("failed to get blob metadata, please retry")
 	}
 
+	if blobMetadata.Expiry < uint64(time.Now().Unix()) {
+		s.metrics.HandleNotFoundRpcRequest("RetrieveBlob")
+		s.metrics.HandleNotFoundRequest("RetrieveBlob")
+		return nil, api.NewNotFoundError("no metadata found for the given batch header hash and blob index")
```
[the remark this answers is not in the document]
Comment (ian-shim) [reply disperser/apiserver/server.go:733]: To support this, a metadata must never get deleted from the metadata store, but we may eventually want to delete expired metadata

[the remark this answers is not in the document]
Comment (jianoaix) [reply disperser/apiserver/server.go:733]: NotFound looks like a right error to return here

**Review by jianoaix — APPROVED** [review]

**Pull request merged**